pkg/orchestrator/gitHubActions_test.go

```diff
@@ -13,21 +13,39 @@ func TestGitHubActions(t *testing.T) {
 		os.Clearenv()
 		os.Unsetenv("GITHUB_HEAD_REF")
 		os.Setenv("GITHUB_ACTIONS", "true")
+		os.Setenv("GITHUB_REF_NAME", "feat/test-gh-actions")
 		os.Setenv("GITHUB_REF", "refs/heads/feat/test-gh-actions")
 		os.Setenv("GITHUB_RUN_ID", "42")
 		os.Setenv("GITHUB_SHA", "abcdef42713")
-		os.Setenv("GITHUB_SERVER_URL", "github.com")
 		os.Setenv("GITHUB_REPOSITORY", "foo/bar")
 
-		p, _ := NewOrchestratorSpecificConfigProvider()
+		p, _ := func() (OrchestratorSpecificConfigProviding, error) {
+			return &GitHubActionsConfigProvider{
+				run: run{
+					HtmlUrl:      "https://github.com/foo/bar/actions/runs/42",
+					RunStartedAt: time.Time{},
+					HeadCommit: struct {
+						Id        string    `json:"id"`
+						Timestamp time.Time `json:"timestamp"`
+					}{
+						// to be filled
+					},
+					Repository: struct {
+						HtmlUrl string `json:"html_url"`
+					}{
+						HtmlUrl: "https://github.com/foo/bar",
+					},
+				},
+			}, nil
+		}()
```

Comment:
Can you make use of httpmock?

https://github.wdf.sap.corp/ContinuousDelivery/piper-library/blob/01777ff4a3e4ec97adeac004fea547db90b1df6d/pkg/sap/xmake/jobFinder_test.go#L37-L52